### PR TITLE
feat(apollo-client): extended sync apollo client

### PR DESCRIPTION
**Pull request by zouxuoz**
Inspired by https://github.com/apollographql/apollo-client/blob/master/packages/apollo-boost/src/index.ts

### CODE REVIEW

File: packages/apollo-client/src/index.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      onAuthSuccess,
+      onAuthError,
+      onIdTokenExpired,
+    });
+
+    const subscriptionLink = new SubscriptionLink();
+
+    const batchHttpLink = new BatchHttpLink({ uri });
+
+    const link = ApolloLink.from([
```
Comment (ej9x) [inline packages/apollo-client/src/index.js]: why not initialize links outside?

File: packages/apollo-client/src/index.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      authLink,
+      subscriptionLink,
+      batchHttpLink,
+    ]);
+
+    super({ cache, link });
+  }
+}
+
+export { EightBaseApolloClient, gql, InMemoryCache };
```
Comment (ej9x) [inline packages/apollo-client/src/index.js]: Where logic of the fetching schema interfaces now?

File: packages/apollo-client/src/index.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    const batchHttpLink = new BatchHttpLink({ uri });
+
+    const link = ApolloLink.from([
+      fileUploadLink,
+      authLink,
+      subscriptionLink,
+      batchHttpLink,
+    ]);
+
+    super({ cache, link });
```
Comment (ej9x) [inline packages/apollo-client/src/index.js]: We should pass default Apollo Client props too.

File: packages/apollo-client/src/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (zouxuoz) [reply packages/apollo-client/src/index.js]: @vorobeez for less count of imports in the source code, inspired by (https://github.com/apollographql/apollo-client/blob/master/packages/apollo-boost/src/index.ts#L16)

Comment (zouxuoz) [reply packages/apollo-client/src/index.js]: @ej9x now it's optional, if user want it - he can pass the custom cache created with fragments schema. Anyway this logic should be placed outside from the Apollo Client.

File: packages/apollo-client/src/index.js (hunk 1 above)
Comment (zouxuoz) [reply packages/apollo-client/src/index.js]: @ej9x because some of the links depends from the Apollo Client config

**Pull request merged**